### PR TITLE
feat(slider): new TEDI-Ready component #13

### LINKED ISSUE #13: [Slider]: New TEDI-Ready component
https://github.com/TEHIK-EE/tedi-design-system/issues/762 

Design: https://www.figma.com/design/jWiRIXhHRxwVdMSimKX2FF/TEDI-READY-2.8.9--work-in-progress-?node-id=17283-87450&t=NphS3xPvxUKc5h4d-4

## DoD
- [x] The component is developed using best practices, conventions, and modern frontend trends
- [x] The component follows **BEM naming conventions** (if applicable)
- [x] The component implementation **matches the Figma design**, including all defined states and interactions
- [x] The component supports **responsiveness** and scales across breakpoints (if applicable)
- [x] The component does not introduce **runtime errors or warnings**
- [x] Linter passes
- [x] Component has export if needed
- [x] The component has **unit tests** with at least **80% code coverage**
- [x] CodeCov passes minimum coverage threshold
- [x] The component has passed **code review**
- [x] The component has passed **Chromatic visual review**
- [x] **Storybook stories** are created:
  - [x] Cover all Figma variations
  - [x] Include edge cases, empty/error states
- [x] Component’s **API (props, slots, events)** is documented
- [ ] **ZeroHeight is updated** with design/development guidelines
- [ ] **External dependencies** used in the component are documented in Confluence with rationale.
- [x] **Semantic commit format** is used (for `semantic-release`)
- [x] External libraries **comply with [validated open source licenses](https://github.com/TEHIK-EE/tedi-design-system/wiki/Definition-of-Done#validated-open-source-component-licences)**
- [x] Component is versioned correctly and ready for release if required
- [ ] Update [statuses page in Zeroheight](https://www.tedi.ee/1ee8444b7/p/300e17-komponentide-staatused)

### PR DESCRIPTION
**Pull request by airikej**
## Summary by CodeRabbit

* **New Features**
  * Added a TEDI-Ready Slider form component: controlled/uncontrolled modes, min/max labels, optional tooltip, helper/feedback, disabled/invalid states, value formatting, addon slot, and accessibility support.
  * Exposed the Slider from the public API.

* **Documentation**
  * Added component reference with full props and usage examples.

* **Styles**
  * Added slider styles including track, thumb, hover/drag/disabled/invalid states.

* **Storybook**
  * Added interactive stories demonstrating slider states and integrations.

* **Tests**
  * Added comprehensive unit and accessibility tests for behaviors and edge cases.

* **Chores**
  * Added optional overlay prop to enable high-frequency positioning updates.

### CODE REVIEW

File: src/tedi/components/form/slider/slider.module.scss [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  border: 1px solid var(--form-slider-border-default);
+  border-radius: var(--form-slider-radius);
+}
+
+.tedi-slider__thumb-anchor {
+  position: absolute;
+  top: 50%;
+  left: calc(
+    var(--form-slider-thumb-size) / 2 + var(--tedi-slider-progress) * (100% - var(--form-slider-thumb-size)) / 100%
+  );
```
[the remark this answers is not in the document]
Comment (airikej) [reply src/tedi/components/form/slider/slider.module.scss:55]: i fixed it as well

**Comment by airikej** [comment]
@coderabbitai resolve

File: src/tedi/components/form/slider/slider.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+          <Col>
+            <Slider {...sharedArgs} id="Default" />
+          </Col>
+        </Row>
+        <Row>
+          <Col lg={2} md={12} className="display-flex align-items-center">
+            <Text modifiers="bold">Hover</Text>
+          </Col>
+          <Col>
+            <Slider {...sharedArgs} id="Hover" />
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.stories.tsx]: Track color for hover and active states don't seem to match Figma.

File: src/tedi/components/form/slider/slider.module.scss [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+  &:active:not(:disabled)::-moz-range-thumb,
+  &.tedi-slider__input--dragging:not(:disabled)::-moz-range-thumb {
+    background: var(--form-slider-thumb-background-active);
+    border-color: var(--form-slider-thumb-border-active);
+  }
+
+  &:focus-visible:not(:disabled)::-webkit-slider-thumb,
+  &.tedi-slider__input--dragging:not(:disabled)::-webkit-slider-thumb {
+    box-shadow: 0 0 0 var(--tedi-borders-01) var(--tedi-neutral-100),
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.module.scss:147]: Focus ring seems to be thicker and darker than in the design.

File: src/tedi/components/form/slider/slider.tsx [hunk 4]
```diff
@@ -0,0 +1,309 @@
...
+  /**
+   * Renders a tooltip above the thumb showing the current value. Shown only while the slider is
+   * hovered, focused, or being dragged. Value is formatted via `valueFormatter`.
+   * @default true
+   */
+  tooltip?: boolean;
+  /**
+   * Node rendered to the right of the slider, typically a NumberField used to edit the same value.
+   */
+  addonRight?: React.ReactNode;
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.tsx:41]: Clicking on the "Set object" button for this arg in Storybook shows this error:

Image

File: src/tedi/components/form/slider/slider.module.scss (hunk 3 above)
Comment (airikej) [reply src/tedi/components/form/slider/slider.module.scss:147]: should be better, for focus border i used `var(--form-slider-thumb-border-focus)`

File: src/tedi/components/form/slider/slider.tsx (hunk 4 above)
Comment (airikej) [reply src/tedi/components/form/slider/slider.tsx:41]: couldnt reproduce, can you try hard refresh on the slider component?

File: src/tedi/components/form/slider/slider.tsx [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+            <span
+              className={styles['tedi-slider__range-label']}
+              aria-hidden={showCurrentValue ? undefined : 'true'}
+              aria-live={showCurrentValue ? 'polite' : undefined}
+            >
+              {rightLabel}
+            </span>
+          )}
+        </div>
+        {addonRight && <div className={styles['tedi-slider__addon']}>{addonRight}</div>}
```
[the remark this answers is not in the document]
Comment (airikej) [reply src/tedi/components/form/slider/slider.tsx:298]: skip for now

File: src/tedi/components/form/slider/slider.tsx (hunk 4 above)
Comment (ly-tempel-bitweb) [reply src/tedi/components/form/slider/slider.tsx:41]: Tried with Empty Cache and Hard Reload, still the same error appears.
I also opened the Slider branch locally and then clicking on addonRight -> Set object results in:
Screenshot 2026-05-04 at 11 06 44

Comment (airikej) [reply src/tedi/components/form/slider/slider.tsx:41]: i just disabled controls for that prop for now, since it accepts react node its not really very functional prop control there anyway, at least the description is there and examples for developers to use

File: src/tedi/components/form/slider/slider.stories.tsx [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+          step={1}
+          value={value}
+          onChange={setValue}
+        />
+      }
+    />
+  );
+};
+
+export const States: Story = {
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.stories.tsx:181]: Maybe move Sliders below state titles?
Image

File: src/tedi/components/form/slider/slider.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+      type: 'figma',
+      url: 'https://www.figma.com/design/jWiRIXhHRxwVdMSimKX2FF/TEDI-READY-2.45.70?node-id=19071-105925&m=dev',
+    },
+  },
+};
+export default meta;
+
+type Story = StoryObj<typeof Slider>;
+
+export const Default: Story = {
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.stories.tsx:36]: Maybe the clickable area could be a bit bigger like in [Material UI Slider](https://mui.com/material-ui/react-slider/)?

https://github.com/user-attachments/assets/aecb8a71-10a0-470a-885f-90bd7dc0c66c

File: src/tedi/components/form/slider/slider.module.scss [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+  width: 100%;
+  height: 100%;
+  padding: 0;
+  margin: 0;
+  appearance: none;
+  background: transparent;
+  border: 0;
+  border-radius: inherit;
+  outline: none;
+
```
Comment (ly-tempel-bitweb) [inline src/tedi/components/form/slider/slider.module.scss:77]: Add `cursor: pointer;` for the track as well like in Material UI Slider?

**Review by ly-tempel-bitweb — APPROVED** [review]
